release_docs/RELEASE.txt

```diff
@@ -147,11 +147,20 @@ Support for new platforms, languages and compilers
 ==================================================
     -
     
-Bug Fixes since HDF5-1.13.3 release
+Bug Fixes since HDF5-1.14.0 release
```

Comment:
Also is this change something we only make at release time, or was it just forgotten?

Reply:
Forgotten. Good catch.